src/porepy/numerics/ad/discretizations.py

```diff
@@ -496,9 +497,10 @@ def _bound_pressure_face_function(
         # The first part is rather involved.
 
         # Get hold of the underlying flux discretization.
-        base_bound_pressure_face = self._discretization.bound_pressure_face.evaluate(
+        base_bound_pressure_face = self._discretization.bound_pressure_face.value(
             self.equation_system
         )
+        assert isinstance(base_bound_pressure_face, np.ndarray)
```

Comment:
Same comment as above, this should be a sps.spmatrix.

Reply:
```suggestion
        # NOTE: This assertion will most likely break, the expected type should be sps.spmatrix.
        # While the assertion could have been changed, this would lead to an error further below.
        # This reflects that the current implementation is incomplete, thus the code should only
        # be used with extreme caution. A reimplementation of the method with differentiable
        # Tpfa is under way, thus fixing the current implementation has not been prioritized.
        assert isinstance(base_bound_pressure_face, np.ndarray)
```
Something like this should give a fair warning. I cannot imagine that flake8 will approve of that formatting, sorry..